### PR TITLE
[SPARK-38692][SQL] Use error classes in the compilation errors of function args

### PR DESCRIPTION
**Pull request by lvshaokang**
### What changes were proposed in this pull request?
Migrate the following errors in QueryCompilationErrors:
- invalidFunctionArgumentsError
- invalidFunctionArgumentNumberError
- functionAcceptsOnlyOneArgumentError
- secondArgumentNotDoubleLiteralError
- functionCannotProcessInputError
- v2FunctionInvalidInputTypeLengthError
- secondArgumentInFunctionIsNotBooleanLiteralError

### Why are the changes needed?
To follow the error class framework and improve test coverage.

### Does this PR introduce _any_ user-facing change?
No.

### How was this patch tested?

```
build/sbt "test:testOnly *QueryCompilationErrorsSuite"
```

### CODE REVIEW

**Comment by LuciferYang** [comment]
cc @MaxGekk

File: sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala [hunk 1]
```diff
@@ -153,6 +153,9 @@ object FunctionRegistryBase {
         } catch {
           // the exception is an invocation exception. To get a meaningful message, we need the
           // cause.
+          case e: Exception if e.getCause.isInstanceOf[AnalysisException] =>
```
Comment (LuciferYang) [inline sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:156]: does this eqs
```scala
case e: Exception =>
    e.getCause match {
       case ae: AnalysisException if ae.errorClass.isDefined => throw ae;
       case other => throw new AnalysisException(other.getMessage)
    }
```

Comment (lvshaokang) [reply sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/FunctionRegistry.scala:156]: @LuciferYang Thanks for you review.

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -105,6 +105,10 @@
     "message" : [ "The fraction of sec must be zero. Valid range is [0, 60]. If necessary set <config> to false to bypass this error. " ],
     "sqlState" : "22023"
   },
+  "INVALID_FUNCTION_ARGUMENTS" : {
+    "message" : [ "Invalid <invalidType> of arguments for function <functionName>. <message>" ],
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:109]: Could you use sub-classes here, see an example in https://github.com/apache/spark/pull/36307

Comment (lvshaokang) [reply core/src/main/resources/error/error-classes.json:109]: @MaxGekk Thanks. I will be use sub-classes to correct it later.

**Comment by MaxGekk** [comment]
@lvshaokang Could you resolve conflicts, please.

**Comment by lvshaokang** [comment]
@MaxGekk PTAL.I have resolved conflicts,thanks.

File: core/src/main/resources/error/error-classes.json [hunk 3]
```diff
@@ -119,6 +119,33 @@
     "message" : [ "The fraction of sec must be zero. Valid range is [0, 60]. If necessary set <config> to false to bypass this error. " ],
     "sqlState" : "22023"
   },
+  "INVALID_FUNCTION_ARGUMENTS" : {
```
[the remark this answers is not in the document]
Comment (lvshaokang) [reply core/src/main/resources/error/error-classes.json:258]: @MaxGekk Sry, i've been busy. PTAL. I format the error class message and move `INVALID_PARAMETER_VALUE` to `INVALID_FUNCTION_ARGUMENTS` as sub-class, beacuse of i feel parameter value belong to function argument scopes

**Comment by lvshaokang** [comment]
@MaxGekk Sry.I recently had time to complete this issue.Please take a look this pr.I unifyed the error class message and moved the `INVALID_PARAMETER_VALUE` to `INVALID_FUNCTION_ARGUMENTS `, because of i think parameter value belong to funcation arguments scope.

**Pull request closed without merging**